src/Disks/IO/AsynchronousBoundedReadBuffer.cpp

```diff
@@ -250,9 +265,8 @@ bool AsynchronousBoundedReadBuffer::nextImpl()
     size_t bytes_read = result.size - result.offset;
     if (bytes_read)
     {
-        /// Adjust the working buffer so that it ignores `offset` bytes.
-        internal_buffer = Buffer(memory.data(), memory.data() + memory.size());
-        working_buffer = Buffer(memory.data() + result.offset, memory.data() + result.size);
+        page_cache_cell = result.page_cache_cell;
```

Comment:
As far as I see, `page_cache_cell` is not used anyhow in `AsynchronousBoundedReadBuffer`, it is only assigned and reset

Reply:
It's a shared_ptr keeping alive the memory to which working_buffer is pointing.